### PR TITLE
Added config parameter to configure app name …

### PR DESCRIPTION
**Pull request by stefanhorning**
used for tagging the rails metrics to be able to distinguish between multiple apps.

### CODE REVIEW

File: lib/influxdb/rails/configuration.rb [hunk 1]
```diff
@@ -65,6 +67,8 @@ class Configuration
         :series_name_for_view_runtimes        => "rails.view",
         :series_name_for_db_runtimes          => "rails.db",
 
+        :rails_app_name => 'my-rails-app',
```
Comment (dmke) [inline lib/influxdb/rails/configuration.rb]: This should either be `nil`, or something like `Rails.application.class.parent_name`.

File: lib/influxdb-rails.rb [hunk 2]
```diff
@@ -83,15 +83,17 @@ def handle_action_controller_metrics(name, start, finish, id, payload)
...
             values: {
               value: controller_runtime,
             },
             tags: {
-              method: method,
-              server: hostname,
+              method:   method,
+              server:   hostname,
+              app_name: app_name
             },
```
Comment (dmke) [inline lib/influxdb-rails.rb]: `app_name` might be blank, so you'll need a `.compact` on the hash

File: lib/influxdb-rails.rb [hunk 3]
```diff
@@ -101,8 +103,9 @@ def handle_action_controller_metrics(name, start, finish, id, payload)
               value: view_runtime,
             },
             tags: {
-              method: method,
-              server: hostname,
+              method:   method,
+              server:   hostname,
+              app_name: app_name
             },
```
Comment (dmke) [inline lib/influxdb-rails.rb]: same here

File: lib/influxdb-rails.rb [hunk 4]
```diff
@@ -112,8 +115,9 @@ def handle_action_controller_metrics(name, start, finish, id, payload)
               value: db_runtime,
             },
             tags: {
-              method: method,
-              server: hostname,
+              method:   method,
+              server:   hostname,
+              app_name: app_name
             },
```
Comment (dmke) [inline lib/influxdb-rails.rb]: and here

**Review by dmke — CHANGES_REQUESTED** [review]
The default value for `app_name` should not be `my-rails-app`. Existing users of this gem, who haven't set this, shouldn't start to wonder where this might have came from.

I'd suggest either to re-use the application class name, or default to a blank value.

**Comment by stefanhorning** [comment]
Yes, was thinking the same thing. Just wanted to quickly check first if this change flies. Will implement the rest tomorrow.

**Comment by stefanhorning** [comment]
Now done, complete with specs.

**Comment by dmke** [comment]
LGTM, will merge it later, when I'm back home.

**Comment by stefanhorning** [comment]
Perfect, thanks!

**Comment by stefanhorning** [comment]
Tests seem to fail for **some** Ruby versions, not sure why as those are unrelated to my changes.

**Comment by dmke** [comment]
Hm, usually restarting the Travis job helps, but in this case it didn't...

(btw. I totally forgot about this, but I'm on it now :-))

**Comment by dmke** [comment]
Interestingly, with the exact same gemset from the [last succeeding spec run](https://travis-ci.org/influxdata/influxdb-rails/jobs/336505417), your changes still fail locally... I'm not sure, *why* this happens (and only for Ruby < 2.4 for Rails 4.0), but I agree, these failures *should* be unrelated to your changes.

**Comment by dmke** [comment]
Got it. `Hash#compact` is undefined. Could've sworn, that existed in Rails 4.0...

**Comment by stefanhorning** [comment]
I see. Should I find another solution or will you fix it?

**Comment by stefanhorning** [comment]
Fixed now, by no longer relying on `#compact`.

**Comment by dmke** [comment]
This is now merged in fae86f0 and c1f144d. The workaround for `Hash#compact` can be found in 6234e1c.

I've not yet released a new gem version, since I started a major refactoring and want to include #22 and #43 as well. Since 055a952 introduced some breaking changes and at least #43 requires more of them, I'm going for a 1.0 release.

**Comment by dmke** [comment]
After some cleanup, I've discovered [`config.application_name`](https://github.com/influxdata/influxdb-rails/blob/master/lib/influxdb/rails/configuration.rb#L27), which actually gets already set to `Rails.application.class.parent_name` [in the Railtie](https://github.com/influxdata/influxdb-rails/blob/master/lib/influxdb/rails/railtie.rb#L18).

I'm going to remove `rails_app_name` from the configuration and use the value of `application_name` instead.

**Pull request closed without merging**